bears/uploadTool.py

```diff
@@ -0,0 +1,71 @@
+import argparse
+import glob
+import os
+import re
+import shutil
+import subprocess
+from coalib.collecting.Importers import iimport_objects
+from string import Template
+
+
+def touch(file_name):
+    """
+    Creates an empty file.
+
+    :param file_name: Name of the file.
+    """
+    if not os.path.exists(file_name):
+        open(file_name, 'w').close()
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Choose to upload'
+                                     'the bears or not')
+    parser.add_argument('-u', '--upload', help='Upload the bears',
+                        action='store_true')
+    args = parser.parse_args()
+
+    os.makedirs('upload', exist_ok=True)
+
+    bears_full_filenames = glob.glob('**/*Bear.py')
+
+    for bear_file_name in bears_full_filenames:
+        file_name = bear_file_name
+        bear_name = re.match(r'.+/(.*)\.py', bear_file_name).group(1)
+        bear_object = next(iimport_objects(file_name, attributes='kind',
+                                           local=True))
+        os.makedirs(os.path.join('upload', bear_name), exist_ok=True)
+        os.makedirs(os.path.join('upload', bear_name, bear_name), exist_ok=True)
+        touch(os.path.join('upload', bear_name, bear_name,
+                           '__init__.py'))
+        shutil.copyfile(file_name, os.path.join('upload', bear_name, bear_name,
+                                                bear_name + '.py'))
+
+        NAME = bear_name
+        VERSION = '1.0.30'  # testing purposes, not sure yet how we hold it
+        AUTHORS = str(bear_object.AUTHORS)
+        AUTHORS_EMAILS = str(bear_object.AUTHORS_EMAILS)
+        MAINTAINERS = str(bear_object.maintainers)
+        MAINTAINERS_EMAILS = str(bear_object.maintainers_emails)
+        PLATFORMS = str(bear_object.PLATFORMS)
+        LICENSE = str(bear_object.LICENSE)
+        LONG_DESCRIPTION = str(bear_object.__doc__)
+
+        with open('setup.py.in') as fl:
+            setup_py = Template(fl.read()).safe_substitute(**locals())
+
+        with open(os.path.join('upload', bear_name,
+                               'setup.py'), 'w') as setup_py_handle:
```

Comment:
btw it's no file-handle, it's an file-object (the handle comes directly from the OS :D)